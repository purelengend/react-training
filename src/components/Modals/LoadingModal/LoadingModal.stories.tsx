import type { Meta, StoryObj } from '@storybook/react';
import LoadingModal from './index';
import './loading-modal.module.css';
const meta: Meta<typeof LoadingModal> = {
  title: 'Example/LoadingModal',
  component: LoadingModal,
  argTypes: {
    isVisible: {
      description: 'Indicates whether the modal is hidden or visible.'
    }
  }
};

type Story = StoryObj<typeof LoadingModal>;

export const LoadingModalExample: Story = {
  args: {
    isVisible: true
  }
};
export default meta;
