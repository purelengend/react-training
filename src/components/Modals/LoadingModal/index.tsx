import { memo } from 'react';
import { Spinner } from '../../common/Spinner';

interface LoadingModalProps {
  isVisible: boolean;
}
const LoadingModal = memo(({ isVisible }: LoadingModalProps) => {
  return (
    isVisible && (
      <div
        style={{ display: isVisible ? 'inline-flex' : 'none' }}
        className="d-flex-center modal-overlay"
      >
        <Spinner />
      </div>
    )
  );
});

export default LoadingModal;
