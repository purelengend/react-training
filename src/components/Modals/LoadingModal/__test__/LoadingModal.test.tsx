import { customRenderer } from '@src/test/test-utils';

import LoadingModal from '..';

describe('LoadingModal test case', () => {
  it('should render correctly', () => {
    const mockLoadingModal = customRenderer(<LoadingModal />);

    expect(mockLoadingModal.toJSON()).toMatchSnapshot();
  });
});
