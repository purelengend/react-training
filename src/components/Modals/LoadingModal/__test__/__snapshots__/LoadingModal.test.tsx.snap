// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingModal test case > should render correctly 1`] = `
<div
  className="d-flex-center modal-overlay"
>
  <div
    className="_loader_3c694b _modal-loading_81953c"
    id="spin"
  />
</div>
`;
